kiwi/volume_manager/btrfs.py

```diff
@@ -438,6 +426,7 @@ def set_property_readonly_root(self):
         root_is_readonly_snapshot = \
             self.custom_args['root_is_readonly_snapshot']
         if root_is_snapshot and root_is_readonly_snapshot:
+            # TODO we could call snapper modify here instead
             sync_target = self.get_mountpoint()
             Command.run(
                 ['btrfs', 'property', 'set', sync_target, 'ro', 'true']
```

Comment:
Here we could run something like `snapper --dbus --root <image_root> modify --read-only 1` if we require snapper as part of the build environment.